.github/workflows/main.yml

```diff
@@ -13,8 +13,17 @@ concurrency:
     cancel-in-progress: ${{ github.ref != 'refs/heads/master' }}
 
 jobs:
+    authorize:
+      name: Authorize
+      environment: ${{ github.event.pull_request.head.repo.fork && 'external' || 'internal' }}
+      runs-on: ubuntu-latest
+      steps:
+        - run: true
```

Comment:
The authorize step is required when u use `pull_request_target` in the `on` on line 3. If u use `pull_request`, authorize has no purpose.

You should ask your self:

- should codecov work for forks? 
  - if yes => use pull_request_target and authorize
  - if no => keep pull_request and drop authorize.

Reply:
Hi @frederikprijck, 

Thank you very much for the detailed explanation on the differences between the triggers `pull_request` and `pull_request_target`.

We would like to have codecov working for forks as well but it needs to be authorized first, hence proceeded with `pull_request_target`.

Pushed the changes accordingly, please take a look again

Reply:
On further discussion with @frederikprijck, I've decided to remove the code coverage check for the PR's raised from forks, hence changing it back to `pull_request` and dropping `authorize`